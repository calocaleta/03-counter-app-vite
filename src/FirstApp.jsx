import React from 'react'

const newMessage = {
    message: 'Hola Mundo',
    title: 'Carlos'
};

const fExp = (valor) => {
    return valor*valor;
}

const FirstApp = () => {
    //const newMessage = 'Nuevo Mensaje!';
    
  return (
    <>
        <h1>{ fExp(3) }</h1>
        <code>{JSON.stringify(newMessage)}</code>
        {/* <code>{JSON.stringify(newMessage)}</code> */}
        <p>Subtitulo</p>
    </>
  )
}

export default FirstApp;