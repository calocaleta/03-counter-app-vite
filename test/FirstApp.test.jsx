/** @jest-environment jsdom */
import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => { 
    
    test('Debe de hacer match con el snapshot',() => {
        
        const title = 'Hola, Soy Goku';
        const {container} = render( <FirstApp title={title} /> );

        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar el tÃ­tulo en un h1',() => {

        const title = 'Hola, Soy Goku';
        const {container, getByText} = render( <FirstApp title={title} /> );

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        epect(h1.innerHTML).toContain(title);


    });

 })