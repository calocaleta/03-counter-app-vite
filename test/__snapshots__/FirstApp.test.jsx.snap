// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
    9
  </h1>
  <code>
    {"message":"Hola Mundo","title":"Carlos"}
  </code>
  <p>
    124
  </p>
</div>
`;

exports[`Pruebas en <FirstApp /> Debe de mostrar el t√≠tulo en un h1 1`] = `
<div>
  <h1>
    Hola, Soy Goku
    9
  </h1>
  <code>
    {"message":"Hola Mundo","title":"Carlos"}
  </code>
  <p>
    124
  </p>
</div>
`;
